enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,216 @@
+---
+title: declarative-index-config
+authors:
+  - "@anik120"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-07-10
+last-updated: 2020-07-10
+status: implementable  
+---
+
+# Declarative Index Config
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions [optional]
+
+## Summary
+
+This enhancement proposes the use of [Cue](https://cuelang.org/docs/about/) to provide a way to declaratively define an index so that 
+operator authors and pipeline developers can reason about the structure of an index in a standard way that's
+also consumable by continuous integration tools. 
+
+## Motivation
+
+Currently, a lot of the rules about the structure of an index are implicitly documented in various sources that are scattered. This makes 
+it hard for operator authors and pipeline developers creating indexes to rapidly make decisions that impact the graph of the index. To alleviate 
+some of the pain points related to that, a number of tools have cropped up (operator-courier, opm etc) to provide users a way to validate their bundles
+and indexes. However, like all tools that aspire to provide data validation, these tools, or api libraries that these tools import rules from, have to 
+be constantly maintained to keep up with the latest rule changes, while also maintaining older rules in previous releases. This introduces a lot of overhead
+in ensuring that olm is easy to use by users, while also leaving windows for lapses due to mis-coordination between different tools across different versions. 
+
+CUE is a configuration and data validation language that makes it easy to define the structure and validate the data with the tools it provides.  
```

Comment:
How does CUE address the need version rule sets and release tooling like `opm` in newer versions that include these newer rule sets?